app/controllers/admins/settings_controller.rb

```diff
@@ -0,0 +1,27 @@
+require 'singleton'
+
+module Admins
+  class SettingsController < BaseController
+    include Singleton
+
+    def update
+      if @settings.update_attributes settings_params
+        redirect_to admin_settings_path
+      else
+        render 'edit'
+      end
+    end
+
+    def set_current_semester
+      @settings.current_semester_id = params[:current_semester_id]
+      @settings.save
+      redirect_to admin_settings_path
```

Comment:
Let's set the flash as well so the user knows an update has been made! Something like

    redirect_to admin_settings_path, notice: "Current semester changed."